### PR TITLE
boards: stm32: Restore Kconfig control on pinmux files

### LINKED ISSUE #24509: Ethernet Sample Echo Failed in Nucleo_f429zi - bisected
**Describe the bug**
I've tried with commit 65eacef644 to use the IP connection on a nucleo_f429zi but it doesn't work.
I bisected it and it seems the problem slipped in with commit 
`commit bfaa1091fd6bc1877992bd40973adfea1412603b
Author: Erwan Gouriou <erwan.gouriou@linaro.org>
Date:   Thu Apr 16 20:52:22 2020 +0200

    boards: stm32: Use dt API for peripheral configuration

    Replace use of Kconfig SPI_X symbols by calls to DT API.

    Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>`

**To Reproduce**
1. cd zephyrproject/zephyr/samples/net/sockets/echo
2. west build -p auto -b nucleo_f429zi .
3. west flash
4. from pc ping 192.0.2.1

**Expected behavior**
Example must work

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

### PR DESCRIPTION
**Pull request by erwango**
Move to new DT API removed the Kconfig control on individual peripheral activation which was used to control board pin configuration and allowed to avoid conflicting usages of pin between peripherals.

While it is good to use dts as a reference for peripheral instances enabling in the sw, a Kconfig notion should be controlled in files that are compiled whatever the use case. 

For all stm32 based boards, add Kconfig subystem control on pin configuration:
 `#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi2), okay) && CONFIG_SPI`

Fixes #24509 (/#25251 (stm32 part))

### CODE REVIEW

**Comment by erwango** [comment]
Removed few people that are not impacted by this change, but I might have removed more than needed, apologies.

File: boards/arm/nucleo_f411re/pinmux.c [hunk 1]
```diff
@@ -34,7 +34,7 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA8, STM32F4_PINMUX_FUNC_PA8_I2C3_SCL},
 	{STM32_PIN_PB4, STM32F4_PINMUX_FUNC_PB4_I2C3_SDA},
 #endif
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi1), okay) || \
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi1), okay) && CONFIG_SPI || \
```
Comment (mbolivar-nordic) [inline boards/arm/nucleo_f411re/pinmux.c:37]: I know && has higher precedence than ||, but this one might be parenthesized just for clarity.

**Review by mbolivar-nordic — APPROVED** [review]
Looks good to me. One suggestion, not blocking

File: boards/arm/nucleo_f411re/pinmux.c (hunk 1 above)
Comment (erwango) [reply boards/arm/nucleo_f411re/pinmux.c:37]: You're right @mbolivar-nordic , fixed.

**Review by mbolivar-nordic — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Comment by carlescufi** [comment]
@erwango #25214 is merged now

**Comment by erwango** [comment]
> @erwango #25214 is merged now

@carlescufi, edited PR desc.

**Pull request merged**